File: src/08.recycle_view/app/src/main/java/com/haidar/myapplication/MainActivity.java
package com.haidar.myapplication;

import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;

import android.view.View;

import android.view.Menu;
import android.view.MenuItem;

import java.util.LinkedList;

public class MainActivity extends AppCompatActivity {

    NavController nav;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        final int[] countdata = {20};
        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                countdata[0]++;

            }
        });

        this.nav = Navigation.findNavController(this,R.id.nav_host_fragment);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }



    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_reset) {

            nav.navigateUp();
            nav.navigate(R.id.FirstFragment);
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: src/learn/app/src/main/java/android/example/com/MainActivity.kt
package android.example.com

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.roll_button)
        val textView: TextView = findViewById<TextView>(R.id.result_text)
        textView.text = "Dice Rolled !"
        rollButton.setOnClickListener { rollDice() }
        add_button.setOnClickListener { add1() }
    }

    private fun rollDice(){
        Toast.makeText(this, "rolled", Toast.LENGTH_SHORT).show()
        val randomInt = (1..6).random()
        result_text.text = randomInt.toString()
    }

    private fun add1(){
        val textView: TextView = findViewById<TextView>(R.id.result_text)
        var res = textView.text.toString()
        if(res.equals("Dice Rolled !")){
            Toast.makeText(this, "value initialized", Toast.LENGTH_SHORT).show()
            textView.text = "1";
        }else if(res.equals("6")){
            Toast.makeText(this, "value is max", Toast.LENGTH_SHORT).show()
            textView.text = "6"
        }else{
            Toast.makeText(this, "added 1", Toast.LENGTH_SHORT).show()
            res = (res.toInt()+1).toString()
            textView.text = res
        }
    }
}